src/analyses/variable-sensitivity/dependency_context_abstract_object.cpp

```diff
@@ -345,6 +356,89 @@ abstract_objectt::locationst
   return existing_locations;
 }
 
+/**
+ * Determine whether 'this' abstract_object has been modified in comparison
+ * to a previous 'before' state.
+ * \param before The abstract_object_pointert to use as a reference to
+ * compare against
+ * \return true if 'this' is considered to have been modified in comparison
+ * to 'before', false otherwise.
+ */
+bool dependency_context_abstract_objectt::has_been_modified(
+  const abstract_object_pointert before) const
+{
+  if(this==before.get())
+  {
+    // copy-on-write means pointer equality implies no modifications
+    return false;
+  }
+
+  // Even if the pointers are different, it maybe that it has changed only
+  // because of a merge operation, rather than an actual write. Given that
+  // this class has knowledge of where writes have occured, use that
+  // information to determine if any writes have happened and use that as the
+  // proxy for whether the value has changed or not.
+  //
+  // For two sets of last written locations to match,
+  // each location in one set must be equal to precisely one location
+  // in the other, since a set can assume at most one match
+  auto before_context=
```

Comment:
I'd be tempted to move this check before the comment since is another edge case we're dealing with early